code/modules/projectiles/guns/energy/special.dm

```diff
@@ -189,6 +189,14 @@
 	else
 		icon_state = initial(icon_state)
 
+/obj/item/weapon/gun/energy/kinetic_accelerator/burn()

```

Comment:
This is snowflake and also you don't need to set the var to delete it, just do if(power_supply) qdel(power_supply)

Just make it happen for all guns